src/devices/Board/RaspberryPiBoard.cs

```diff
@@ -520,6 +615,215 @@ public override PinUsage DetermineCurrentPinUsage(int pinNumber)
             return PinUsage.Unknown;
         }
 
+        /// <summary>
+        /// Checks if the I2C overlay is activated in the configuraztion file.
+        /// </summary>
+        /// <returns>True if it is.</returns>
+        public bool IsI2cActivated()
+        {
+            _activateI2c.Clear();
+            // We are checking possible activation from here: https://github.com/dotnet/iot/blob/main/Documentation/raspi-i2c.md
+            var config = File.ReadAllText(ConfigurationFile).Split('\n').Where(m => !m.Trim().StartsWith("#")).Where(m => m.Length > 1);
+
+            foreach (var possibleActivation in _possibleI2cActivations)
+            {
+                if (config.Any(m => m.Trim().StartsWith(possibleActivation)))
+                {
+                    _activateI2c.Add(possibleActivation);
+
+                }
+            }
+
+            return _activateI2c.Any();
+        }
+
+        /// <summary>
+        /// Gets the overlay pin assignment for I2C.
+        /// </summary>
+        /// <param name="busId">Bus Id.</param>
+        /// <returns>The set of pins for the given I2C bus.</returns>
+        public int[] GetOverlayPinAssignmentForI2c(int busId)
+        {
+            int scl = -1;
+            int sda = -1;
+            if (!IsI2cActivated())
+            {
+                return new int[0];
+            }
+
+            // Checks if there is an overlay because it does override the default configuration
+            var dtoverlay = _activateI2c.Where(m => m.StartsWith($"dtoverlay=i2c{busId}")).FirstOrDefault();
+            if (!string.IsNullOrEmpty(dtoverlay))
+            {
+                // we have an overlay, extract the pins and check them
+                // dtoverlay=i2c1,pins_2_3
+                var pins = dtoverlay.Substring(dtoverlay.IndexOf('_') + 1).Split('_');
```

Comment:
In the default case (when no pins are given), this will fail. So if `dtoverlay.IndexOf('_')` returns -1, the else case must be used (call `GetDefaultPinAssignmentForI2c()`)